engine/consensus/matching/engine.go

```diff
@@ -262,30 +255,53 @@ func (e *Engine) onReceipt(originID flow.Identifier, receipt *flow.ExecutionRece
 		return nil
 	}
 
-	err = e.ensureStakedNodeWithRole(receipt.ExecutorID, head, flow.RoleExecution)
+	err = e.receiptValidator.Validate(receipt)
```

Comment:
This is a good point Leo. Adding some thoughts here for further discussion:

* the `matching.Engine` re-requests receipts for finalized blocks that haven't been sealed yet (provided the backlog of unsealed finalized blocks is large enough) https://github.com/onflow/flow-go/blob/32081ff9f91d24fd5c10ab57bdf447a70c7dd4ef/engine/consensus/matching/engine.go#L891-L912
* So under the assumption that _all_ Execution nodes are honest and _always_ produce the correct result, I am thinking/hoping that it would be fine to drop Receipts for unknown blocks for the following reason:
  - having a single receipt for a block is sufficient for sealing liveness
  - if a consensus node does not have a receipt, it will eventually request it.  
* Caveat: Execution nodes do not always produce a correct result.
   - Therefore, we ideally need the ability to retrieve receipts from _all_ Execution Nodes. (as we might still miss the correct receipt). We can achieve this extending the code to request receipts for unsealed finalized blocks from all Execution Nodes from which we don't have a receipt yet.  Specifically adjusting this code: https://github.com/onflow/flow-go/blob/32081ff9f91d24fd5c10ab57bdf447a70c7dd4ef/engine/consensus/matching/engine.go#L882-L889

Reply:
I think we can also rely on other consensus nodes to relay receipts. This is already the case because If a consensus node does not have a receipt, it is likely that another consensus node will have it and will include it in the chain, thereby indirecly sharing it with other consensus nodes.

Reply:
OK, agreed to drop it and rely on the `requestPending`. And if so, we'd better add a comment to explain:
```suggestion
        // if the receipt is not valid, because the parent result is unknown, we will drop this receipt.
        // in the case where a child receipt is dropped because it is received before its parent receipt, we will
        // eventually request the parent receipt with the `requestPending` function
	err = e.receiptValidator.Validate(receipt)
```